replica/apply_mutation.cc

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "replica/apply_mutation.hh"
+#include "mutation.hh"
+#include "frozen_mutation.hh"
+#include "log.hh"
+
+extern logging::logger dblog;
+
+namespace replica {
+
+apply_mutation::apply_mutation(const mutation& m)
+    : _schema(m.schema())
+    , _mp(&m)
+{}
+
+apply_mutation::apply_mutation(mutation&& m)
+    : _schema(m.schema())
+    , _m(make_foreign(std::make_unique<const mutation>(std::move(m))))
+    , _mp(_m.get())
+{}
+
+apply_mutation::apply_mutation(schema_ptr s, const frozen_mutation& fm)
+    : _schema(std::move(s))
+    , _fmp(&fm)
+{}
+
```

Comment:
Such trivial constructors are better inline. And it's a waste to have a source file with hundreds of thousand of included lines, just for 100 source lines.

Reply:
In a later patch it is beefed up a bit.